### DISCUSSION TITLE
Using Synthea for Intensive Care data

### DISCUSSION THREAD

**Discussion by Iain-S** [synthetichealth/synthea#1062] [Q&A]
@jawalonoski,

I would be interested in using Synthea to generate some synthetic intensive care data. I had a couple of questions such as:

1. If we're only interested in parts of a patient's records when they were in ICU, what is the best way to filter out other data? Is it best to generate the full, usual output, load the result, look for "Admit to ICU" events and drop anything outside of those time widows?
2. It doesn't look like the observations, encounters, etc. have a location associated with them. We were hoping to covert the .csv results to OMOP format with <https://ohdsi.github.io/ETL-Synthea/> but would the approach, above, require us to parse the .fhir output files instead?

I was wondering whether you or a colleague might be able to spare half an hour for a video call to discuss or whether it'd be better to keep it on GitHub?

Cheers

Iain

**Comment by jawalonoski** [comment]
## ICU

I did a quick search for "ICU", "intensive", and "Admission" within the modules, which resulted in a few qualifying events...

Module | Type | Code | Description
---|---|---|---
`acute_myeloid_leukemia` | `Procedure` | SNOMED `305351004` | `Admit to intensive care unit (ICU)`
`congestive_heart_failure` | `Encounter` | SNOMED `305351004` | `Admission to intensive care unit (procedure)`
`sepsis` | `Procedure` | SNOMED `305351004` | `Admit to ICU (procedure)`
`covid19` | `Encounter` | SNOMED `305351004` | `Admission to intensive care unit (procedure)`

So, we have sparse coverage of ICU data.

Also, it is clear we do not consistently use the same display value for the same code, nor are we consistent about whether it is a `Procedure` or an `Encounter`.

## Q & A

> 1. If we're only interested in parts of a patient's records when they were in ICU, what is the best way to filter out other data? Is it best to generate the full, usual output, load the result, look for "Admit to ICU" events and drop anything outside of those time widows?

My usual advice is to generate a full set of output and filter the data to what you want.

You could try to use a [Keep Module](https://github.com/synthetichealth/synthea/wiki/Keep-Patients-Module) if you are comfortable creating one with the [Module Builder](https://synthetichealth.github.io/module-builder/), but since our current logic does not allow you to check for previous encounters or procedures with specific codes, you would have to key off something else, such as an Active Condition or an attribute (i.e., variable) on the patient.

**Do not use the [M Feature](https://github.com/synthetichealth/synthea/wiki/The--M-Feature).**

> 2. It doesn't look like the observations, encounters, etc. have a location associated with them. We were hoping to covert the .csv results to OMOP format with https://ohdsi.github.io/ETL-Synthea/ but would the approach, above, require us to parse the .fhir output files instead?

In the CSV data, observations and all other data are always conducted within and associated with an Encounter.

An Encounter takes place with an Organization and a Provider.

Both Organization and Provider have a location as an address and latitude/longitude.

> I was wondering whether you or a colleague might be able to spare half an hour for a video call to discuss or whether it'd be better to keep it on GitHub?

Either way works, but from your profile it looks like we're in different timezones. We usually decline meetings from the general public (due to time more than anything else), but you appear to be associated with a legitimate research organization, so if you'd like to arrange a meeting, please reach out using our email list: <synthea-list@groups.mitre.org>

**Reply by Iain-S** [reply]
Thanks for the answers. I'll drop an email to the list you mentioned.
